### PR TITLE
Update from Primer React v20 to v30

### PR DESCRIPTION
**Pull request by colebemis**
This PR upgrades our Primer React dependency from v20 to v30. The bulk of the work done in this PR migrating from deprecated color variables to the new [functional color system variables](https://primer.style/primitives/color). Note: upgrading the new color system results in slight visual changes (similar to the changes we saw when migrating github.com to the new functional color system):

| Before | After |
| --- | --- |
| CleanShot 2021-09-30 at 15 44 29@2x | CleanShot 2021-09-30 at 15 46 34@2x |

This upgrade lays the groundwork for supporting dark mode documentation in the future.

## TODO
- [x] Test in primer.style/react
- [x] Test in primer.style/css
- [x] Test in primer.style/view-components

Closes https://github.com/primer/doctocat/issues/253

### CODE REVIEW

File: .changeset/gentle-turtles-cheat.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@primer/gatsby-theme-doctocat": major
+---
+
+Update from `@primer/components` v20.0.0 to v30.0.0
```
Comment (colebemis) [inline .changeset/gentle-turtles-cheat.md:5]:
```suggestion
Update from `@primer/components` v20.0.0 to v30.0.0 which includes a migration to the new [functional color variables](https://primer.style/primitives/color).
```

**Review by jonrohan — APPROVED** [review]
A lot to review here, you could separate out the octicon update first. Then again, if it builds and deploys to vercel ok then. 👍🏻

File: docs/package.json [hunk 2]
```diff
@@ -9,8 +9,7 @@
     "now-build": "npm run build"
   },
   "dependencies": {
-    "@primer/components": "^20.0.0",
-    "@primer/css": "^12.4.1",
```
[the remark this answers is not in the document]
Comment (colebemis) [reply docs/package.json:13]: @primer/css wasn't ever used. Not sure why it was a dependency

**Comment by colebemis** [comment]
@jonrohan The octicons update required 2 lines of code to be changed. The rest of this is color system work + replacing deprecated utility components.

**Comment by colebemis** [comment]
I tested in primer.style/react it worked. I'll test it in primer.style/css and /view-components too.

**Pull request merged**